x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java

```diff
@@ -296,13 +322,44 @@ protected XPackLicenseState getLicenseState() {
         // the transform services should have been created
         assert transformServices.get() != null;
 
-        return Collections.singletonList(new TransformPersistentTasksExecutor(client, transformServices.get(), threadPool, clusterService,
-            settingsModule.getSettings(), expressionResolver));
+        return Collections.singletonList(
+            new TransformPersistentTasksExecutor(
+                client,
+                transformServices.get(),
+                threadPool,
+                clusterService,
+                settingsModule.getSettings(),
+                expressionResolver
+            )
+        );
     }
 
     @Override
     public List<Setting<?>> getSettings() {
-        return Collections.singletonList(NUM_FAILURE_RETRIES_SETTING);
+        return Collections.unmodifiableList(Arrays.asList(TRANSFORM_ENABLED_NODE, NUM_FAILURE_RETRIES_SETTING));
+    }
+
+    @Override
+    public Settings additionalSettings() {
+        String transformEnabledNodeAttribute = "node.attr." + TRANSFORM_ENABLED_NODE_ATTR;
+        String transformRemoteEnabledNodeAttribute = "node.attr." + TRANSFORM_REMOTE_ENABLED_NODE_ATTR;
+
+        if (enabled == false) {
+            return Settings.EMPTY;
+        }
+
+        if (settings.get(transformEnabledNodeAttribute) != null || settings.get(transformRemoteEnabledNodeAttribute) != null) {
+            throw new IllegalArgumentException(
+                "Directly setting transform node attributes is not permitted, please use the documented node settings instead"
```

Comment:
In the ML plugin we ban these node attributes even when the plugin is disabled.  In other words the `enabled == false` check comes after the check that throws this error.  I think it's best because otherwise you can get the weird situation where somebody sets the node attribute directly when the plugin is disabled on a node and then other nodes in the cluster might assign transform persistent tasks to the node.  Such tasks wouldn't do anything because no plugin would be watching for them on the node they're assigned to.

(Of course, this is an extreme edge case and unlikely to ever happen.  It also ties in with the conversation about whether it should be permitted to disable plugins on just a subset of nodes.  Moving to a world where transforms is always enabled will simplify things a lot.)